src/lib/ui/thermostat/TemperatureControlGroup.tsx

```diff
@@ -0,0 +1,164 @@
+}
+
+export function TemperatureControlGroup({
+  mode,
+  heatValue,
+  onHeatValueChange,
+  coolValue,
+  onCoolValueChange,
+  delta,
+  minHeat,
+  maxHeat,
+  minCool,
+  maxCool,
+}: TemperatureControlGroupProps): JSX.Element {
+  const showHeat = mode === 'heat' || mode === 'heat_cool'
+  const showCool = mode === 'cool' || mode === 'heat_cool'
+
+  const getHeatBounds = useCallback(() => {
+    const actualMaxHeat =
+      mode === 'heat_cool'
+        ? Math.min(maxHeat - delta, maxCool - delta)
+        : maxHeat
+
+    const actualMinHeat =
+      mode === 'heat_cool' ? Math.max(minHeat, minCool) : minHeat
+
+    return { min: actualMinHeat, max: actualMaxHeat }
+  }, [mode, minHeat, maxHeat, minCool, maxCool, delta])
+
+  const getCoolBounds = useCallback(() => {
+    const actualMaxCool =
+      mode === 'heat_cool' ? Math.min(maxCool, maxHeat) : maxCool
+
+    const actualMinCool =
+      mode === 'heat_cool'
+        ? Math.max(minCool + delta, minHeat + delta)
+        : minCool
+
+    return { min: actualMinCool, max: actualMaxCool }
+  }, [mode, minCool, maxCool, minHeat, maxHeat, delta])
+
+  const adjustHeat = useCallback(
+    (temperature: number) => {
+      const { min, max } = getHeatBounds()
+      const adjustedHeat = Math.min(Math.max(temperature, min), max)
+
+      onHeatValueChange(adjustedHeat)
+
+      if (coolValue < adjustedHeat + delta) {
+        onCoolValueChange(adjustedHeat + delta)
+      }
+    },
+    [getHeatBounds, coolValue, delta, onHeatValueChange, onCoolValueChange]
+  )
+
+  const adjustCool = useCallback(
+    (temperature: number) => {
+      const { min, max } = getCoolBounds()
+      const adjustedCool = Math.min(Math.max(temperature, min), max)
+
+      onCoolValueChange(adjustedCool)
+
+      if (heatValue > adjustedCool - delta) {
+        onHeatValueChange(adjustedCool - delta)
+      }
+    },
+    [getCoolBounds, heatValue, delta, onCoolValueChange, onHeatValueChange]
+  )
+
+  useEffect(() => {
+    if (mode === 'heat_cool') {
+      const { min: minHeat, max: maxHeat } = getHeatBounds()
+      const { min: minCool, max: maxCool } = getCoolBounds()
+
+      if (heatValue < minHeat) {
+        onHeatValueChange(minHeat)
+      } else if (heatValue > maxHeat) {
+        onHeatValueChange(maxHeat)
+      }
+
+      if (coolValue < minCool) {
+        onCoolValueChange(minCool)
+      } else if (coolValue > maxCool) {
+        onCoolValueChange(maxCool)
+      }
+    }
+
+    if (mode === 'heat') {
+      const { min: minHeat, max: maxHeat } = getHeatBounds()
+
+      if (heatValue < minHeat) {
+        onHeatValueChange(minHeat)
+      } else if (heatValue > maxHeat) {
+        onHeatValueChange(maxHeat)
+      }
+    }
+
+    if (mode === 'cool') {
+      const { min: minCool, max: maxCool } = getCoolBounds()
+
+      if (coolValue < minCool) {
+        onCoolValueChange(minCool)
+      } else if (coolValue > maxCool) {
+        onCoolValueChange(maxCool)
+      }
+    }
```

Comment:
These conditions should always be mutually exclusive, so the `else` part is not needed here.
```suggestion
      if (heatValue < minHeat) {
        onHeatValueChange(minHeat)
      }
      if (heatValue > maxHeat) {
        onHeatValueChange(maxHeat)
      }

      if (coolValue < minCool) {
        onCoolValueChange(minCool)
      }
      if (coolValue > maxCool) {
        onCoolValueChange(maxCool)
      }
    }

    if (mode === 'heat') {
      const { min: minHeat, max: maxHeat } = getHeatBounds()

      if (heatValue < minHeat) {
        onHeatValueChange(minHeat)
      }
      if (heatValue > maxHeat) {
        onHeatValueChange(maxHeat)
      }
    }

    if (mode === 'cool') {
      const { min: minCool, max: maxCool } = getCoolBounds()

      if (coolValue < minCool) {
        onCoolValueChange(minCool)
      } 
      if (coolValue > maxCool) {
        onCoolValueChange(maxCool)
      }
    }
```